pointmatcher/Registrar.h

```diff
@@ -90,80 +90,79 @@ namespace PointMatcherSupport
-				
+
 				// check that all parameters were set
-				for (BOOST_AUTO(it, params.begin()); it != params.end() ;++it)
+				for (const auto& param : params)
 				{
-					if (instance->parametersUsed.find(it->first) == instance->parametersUsed.end()){
+					if (instance->parametersUsed.find(param.first) == instance->parametersUsed.end()){
 						throw Parametrizable::InvalidParameter(
-							(boost::format("Parameter %1% for module %2% was set but is not used") % it->first % className).str()
+							(boost::format("Parameter %1% for module %2% was set but is not used") % param.first % className).str()
 						);
 					}
 				}
-				
 				return instance;
 			}
+
 			virtual const std::string description() const
 			{
 				return C::description();
 			}
+
 			virtual const Parametrizable::ParametersDoc availableParameters() const
 			{
 				return C::availableParameters();
 			}
 		};
-		
+
 		//! A descriptor for a class C that does not provide any parameter
 		template<typename C>
 		struct GenericClassDescriptorNoParam: public ClassDescriptor
 		{
 			virtual std::shared_ptr<Interface> createInstance(const std::string& className, const Parametrizable::Parameters& params) const
 			{
-				for (BOOST_AUTO(it, params.begin()); it != params.end() ;++it)
+				for (const auto& param : params)
 					throw Parametrizable::InvalidParameter(
-							(boost::format("Parameter %1% was set but module %2% dos not use any parameter") % it->first % className).str()
+							(boost::format("Parameter %1% was set but module %2% dos not use any parameter") % param.first % className).str()
 						);
+
 				return std::make_shared<C>();
 			}
+
 			virtual const std::string description() const
 			{
 				return C::description();
 			}
+
 			virtual const Parametrizable::ParametersDoc availableParameters() const
 			{
 				return Parametrizable::ParametersDoc();
 			}
 		};
-		
+
 	protected:
-		typedef std::map<std::string, ClassDescriptor*> DescriptorMap; //!< descriptors for sub-classes of Interface, indexed by their names
+		typedef std::map<std::string, std::shared_ptr<ClassDescriptor>> DescriptorMap; //!< descriptors for sub-classes of Interface, indexed by their names
 		DescriptorMap classes; //!< known classes that can be constructed
-		
+
 	public:
-		//! Destructor, remove all classes descriptors 
-		~Registrar()
-		{
-			for (BOOST_AUTO(it, classes.begin()); it != classes.end(); ++it)
-				delete it->second;
-		}
 		//! Register a class by storing an instance of a descriptor helper class
-		void reg(const std::string &name, ClassDescriptor* descriptor)
+		void reg(const std::string &name, std::shared_ptr<ClassDescriptor> descriptor)
 		{
 			classes[name] = descriptor;
 		}
-		
+
 		//! Return a descriptor following a name, throw an exception if name is invalid
-		const ClassDescriptor* getDescriptor(const std::string& name) const
+		std::shared_ptr<ClassDescriptor> getDescriptor(const std::string& name) const
```

Comment:
This type is mentioned in line 156. Does it make sense to create an alias? That way we also abstract a bit the type of smart pointer and the library of origin.

Reply:
Like I said, I'm not enough familiar with the library to tell what is the best thing to do. I've searched in the code if there was such an alias definition for a shared pointer and I haven't found any. What alias would you suggest for this type?

@pomerlef and @simonpierredeschenes What do you think?

Reply:
This item is a bit more of a nitpick, I'd say :slightly_smiling_face: 

Feel free to address it or discard it.

Reply:
No problems! I'll leave it like that then. Thanks!